src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -189,57 +303,139 @@ public String addEmailAlias(String alias) throws Exception
     }
 
     /**
-     * Creates a random document based on {@link DocumentType} passed
-     * Return the {@link Document} object on success creation
+     * It will create a new file in current resource
      * <code>
-     * dataContent.usingUser(userModel).usingResource(myFolder).createContent(DocumentType.TEXT_PLAIN);
-     * </code>
-     * 
-     * @param documentType
+     * FileModel newRandomFile = FileModel.getRandomFileModel();
+     * dataContent.usingUser(testUser).usingSite(testSite).createContent(newRandomFile);
+     * <code>
+     */
+    public FileModel createContent(){
+        return createContent(new FileModel(RandomData.getRandomName("File")));
+    }
+
+    /**
+     * This is the entry point of the createContent() method to make REST API or CMIS call
+     */
+    public FileModel createContent(FileModel fileModel){
+        AlfrescoHttpClient client = alfrescoHttpClientFactory.getObject();
+
+        if(client.getAlfVersion() >= 5.2)
+        {
+            FileModel createFile = createContentV1Api(client, fileModel);
+            this.updateContent(client, createFile);

```

Comment:
what happens if the createContentV1Api method does not create the file ? how is this handled in the updateContent method?  how is it handled by the calling code for createContent method ?